mm2src/coins/eth.rs

```diff
@@ -913,15 +913,16 @@ macro_rules! tx_type_from_pay_for_gas_option {
 impl EthCoinImpl {
     #[cfg(not(target_arch = "wasm32"))]
     fn eth_traces_path(&self, ctx: &MmArc, my_address: Address) -> PathBuf {
-        ctx.dbdir()
+        ctx.address_dir(&my_address.addr_to_string())
```

Comment:
nit: I personally prefer passing strict types into functions to ensure validation within them.

Reply:
im keeping it simple as `address_dir()` is used in other places in which it might not be easy to construct a full `Address` struct (or enum of possible address types actually) from a string.
e.g. it's simpler when dealing with addresses fetched as strings from a DB in a case where u have no easy access to `Coin` to convert that string to `Address` or maybe no coins are enabled yet.

Reply:
Maybe make it `T: ToString` and pass the `Address` where possible? That way we can at least ensure validated usage in most cases I suppose.

Reply:
> Maybe make it T: ToString and pass the Address where possible?

note, that you shouldn't call to_string for eth Address type, it will be truncated and look like "123...456"

Reply:
@mariocynicys I updated my comment above ^ with clarification